### ISSUE TITLE
[BUG] NgramFwBwPerplexityMetric and BleuFwBw should set indicates test set instead of key

### ISSUE THREAD

**Issue by hzhwcmhf** [thu-coai/cotk#269]
There's some problem that we can't use reference_test_key in these Metrics:
1. what if I want to use 'train' or 'dev' for inference?
2. the key is confusing because we don't tell the user the key of dataloader._data

Here's a solution:
1. Pass a 2-d jagged list to NgramFwBwPerplexityMetric.__init__, indicating the reference set.
2. Add ``get_all_batch(key)`` to dataloader, it should returned a dict with the same key value pair as ``get_batch``, except all the values should be jagged array. (Some repeated key value pair like "sent" can be ommited, because "sent_allvocabs" have the similar information and it is used for metrics.)

Attention:
Docstring should be maintained.
